File: initials-project-ios/Utility/InputProcessor.m
//
//  InputProcessor.m
//  initials-project-ios
//
//  Created by Vladislav Kleschenko on 5/11/18.
//  Copyright © 2018 Uladzislau Kleshchanka. All rights reserved.
//

#import "InputProcessor.h"

@implementation InputProcessor

- (instancetype)initWithPattern:(NSString *)pattern {
    if (self = [super init]) {
        _pattern = pattern;
    }
    return self;
}

- (NSArray<NSString *> *)initials {
    NSMutableArray *initials = [NSMutableArray array];
    if (self.words != nil) {
        for (NSString *word in self.words) {
            [initials addObject:[[word substringToIndex:1] capitalizedString]];
        }
    }
    return initials;
}

- (void)processInput:(NSString *)text {
    NSCharacterSet *separators = [NSCharacterSet characterSetWithCharactersInString:self.pattern];
    _words = [[text stringByTrimmingCharactersInSet:[NSCharacterSet characterSetWithCharactersInString:@" "]] componentsSeparatedByCharactersInSet:separators];
}

- (NSInteger)count {
    return self.words.count;
}
@end

File: initials-project-ios/Utility/InputProcessor.swift
//
//  InputProcessor.swift
//  initials-project-ios
//
//  Created by Uladzislau Kleshchanka on 6/15/18.
//  Copyright © 2018 Uladzislau Kleshchanka. All rights reserved.
//

import Foundation

@objc public class InputProcessor: NSObject {
    private let characterSet: CharacterSet
    private(set) var initials = [String]()
    
    init(withPattern pattern: String) {
        characterSet = CharacterSet.init(charactersIn: pattern)
    }
    
    func processInput(_ text: String) {
        let words = text.components(separatedBy: characterSet)
        for word in words {
            guard let char = word.first else {continue}
            initials.append(String(char).capitalized)
        }
    }
}

